7.38.0 changelog (#12852)
DataDog/datadog-agent: CHANGELOG.rst

Release Notes
=============

.. _Release Notes_7.38.0:

7.38.0 / 6.38.0
======

.. _Release Notes_7.38.0_Prelude:

Prelude
-------

Release on: 2022-07-25

- Please refer to the `7.38.0 tag on integrations-core <https://github.com/DataDog/integrations-core/blob/master/AGENT_CHANGELOG.md#datadog-agent-version-7380>`_ for the list of changes on the Core Checks

.. _Release Notes_7.38.0_New Features:

New Features
------------

> - Add `otlp_config.debug.loglevel` to determine log level when the OTLP Agent receives metrics/traces for debugging use cases.

Reviewer: Let's move this to Enhancements

```suggestion
```

  and show on the status page. 

- Add a `containerd_exclude_namespaces` configuration option for the Agent to
  ignore containers from specific containerd namespaces.

- The `log_level` of the agent is now appended
  to the flare archive name upon its creation.

- The metrics reported by KSM core now include the tags "kube_app_name",
  "kube_app_instance", and so on, if they're related to a Kubernetes entity
  that has a standard label like "app.kubernetes.io/name",
  "app.kubernetes.io/instance", etc.

- The Kubernetes State Metrics Core check now collects two ingress metrics:
  ``kubernetes_state.ingress.count`` and ``kubernetes_state.ingress.path``.

- Move process chunking code to util package to avoid cycle import when using it in orchestrator check.

- APM: Add support for PostgreSQL JSON operators in the SQL obfuscate package.

- The OTLP ingest endpoint now supports the same settings and protocol as the OpenTelemetry Collector OTLP receiver v0.54.0 (OTLP v0.18.0).

- The Agent now embeds Python-3.8.13, an upgrade from
  Python-3.8.11.

- APM: Updated Rare Sampler default configuration values to sample traces more uniformly across environments and services.

> - Support Exponential Histograms with delta aggregation temporality.

Reviewer: ```suggestion
- The OTLP ingest endpoint now supports Exponential Histograms with delta aggregation temporality.
```

- The Windows installer now supports grouped Managed Service Accounts.

> - Enable https monitoring on arm64 with kernel >= 5.5.0.

Reviewer: ```suggestion
- Enable https monitoring on arm64 with kernel >= 5.5.0.

- Add ``otlp_config.debug.loglevel`` to determine log level when the OTLP Agent receives metrics/traces for debugging use cases.
```

.. _Release Notes_7.38.0_Deprecation Notes:

Deprecation Notes
-----------------

> - Use `instrumentation_scope_metadata_as_tags` instead of `instrumentation_library_metadata_as_tags`
>   in favor of https://github.com/open-telemetry/opentelemetry-proto/releases/tag/v0.15.0
>   Both must not be set at the same time.

Reviewer: ```suggestion
- Deprecate``otlp_config.metrics.instrumentation_library_metadata_as_tags`` in 
  in favor of ``otlp_config.metrics.instrumentation_scope_metadata_as_tags``.
```
